From a50330e7638603d13ff031bfaeaa1ca9c885431c Mon Sep 17 00:00:00 2001
From: Jiri Olsa <jolsa@redhat.com>
Date: Mon, 25 Aug 2014 07:50:10 -0400
Subject: [tools] lib/traceevent: Get rid of malloc_or_die() in pevent_filter_add_filter_str()

Message-id: <1408953210-26343-123-git-send-email-jolsa@redhat.com>
Patchwork-id: 88917
O-Subject: [PATCH RHEL7.1 BZ1133083 122/322] tools lib traceevent: Get rid of malloc_or_die() in pevent_filter_add_filter_str()
Bugzilla: 1133083
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>

From: Jiri Olsa <jolsa@kernel.org>

Bugzilla: 1133083
https://bugzilla.redhat.com/show_bug.cgi?id=1133083

upstream
========
commit 28942c87e5e907f591d77547203e86ad1089b499
Author: Namhyung Kim <namhyung@kernel.org>
Date: Mon Dec 9 14:34:08 2013 +0900
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/tools/lib/traceevent/parse-filter.c b/tools/lib/traceevent/parse-filter.c
index 53e48eb..a4d5bb2 100644
--- a/tools/lib/traceevent/parse-filter.c
+++ b/tools/lib/traceevent/parse-filter.c
@@ -1226,7 +1226,13 @@ int pevent_filter_add_filter_str(struct event_filter *filter,
   else
    len = strlen(filter_str);
 
-  this_event = malloc_or_die(len + 1);
+  this_event = malloc(len + 1);
+  if (this_event == NULL) {
+   show_error(error_str, "Memory allocation failure");
+   /* This can only happen when events is NULL, but still */
+   free_events(events);
+   return -1;
+  }
   memcpy(this_event, filter_str, len);
   this_event[len] = 0;
 
-- 
1.7.1